src/Compilers/CSharp/Portable/Syntax/CompilationUnitSyntax.cs

```diff
@@ -19,21 +20,56 @@ public IList<ReferenceDirectiveTriviaSyntax> GetReferenceDirectives()
 
         internal IList<ReferenceDirectiveTriviaSyntax> GetReferenceDirectives(Func<ReferenceDirectiveTriviaSyntax, bool>? filter)
         {
+            if (!this.ContainsDirectives)
+                return SpecializedCollections.EmptyList<ReferenceDirectiveTriviaSyntax>();
+
             // #r directives are always on the first token of the compilation unit.
             var firstToken = (SyntaxNodeOrToken)this.GetFirstToken(includeZeroWidth: true);
-            return firstToken.GetDirectives<ReferenceDirectiveTriviaSyntax>(filter);
+            return firstToken.GetDirectives(filter);
```

Comment:
Was the type argument here unnecessary to begin with?

Reply:
Yup.  It was unnecessary from the start.